heroku/research/recommendation-of-adoption.md

```diff
@@ -316,6 +328,63 @@ projects where the development variables are not secret, we can even keep
 this file unencrypted.) In staging and production, secret values will be configured
 in Heroku as config vars for the application.
 
+### CI/CD
+
+In switching away from EC2 as our compute platform, we will no longer be
+able to make use of CodeDeploy for continuous deployment.
+
+Heroku has its own CI offering, [Heroku CI](https://devcenter.heroku.com/articles/heroku-ci),
+that offers concurrent test runners and integration with Heroku Pipelines for
+continuous deployment. Heroku also offers a [GitHub
+integration](https://devcenter.heroku.com/articles/github-integration#automatic-deploys)
+that can build Pipeline stages based on specific branches and that can wait to build until
+external CI (such as Travis) is green.
+
+We recommend moving forward with the GitHub integration for Heroku. This will allow
+us to maintain our current Travis practices for CI, while permitting Heroku to take
+over CD and eliminate the need for deployment scripts.
+
+### Static file management
+
+Containerizing deployments with any deploy provider poses two challenges to
+static file management:
+
+1. Static files cannot be persisted on disk
+2. Nginx cannot serve static files
+
+We offer guidance on these two problems below.
+
+#### 1. Static files cannot be persisted on disk
+
+Since containers have ephemeral filesystems by default, all of the containerization
+providers we considered will require our applications to no longer assume that
+persistent storage is available on disk.
+
+Our deployment practices already assume that an application directory will be fully
+recreated during every deployment, so a change to ephemeral storage should not
+represent a major shift in paradigm. However, the two exceptions to this pattern are
+user-uploaded media and translation files, both of which we [persist on disk
+indefinitely](https://github.com/datamade/deploy-a-site/blob/master/File-uploads.md).
+
+In order to persist user-uploaded media and translation files, we will need to
+make use of an external static file host. Heroku recommends persisting these
+types of files to AWS S3, and [offers documentation on doing
+so](https://devcenter.heroku.com/articles/s3-upload-python). In Django, we will
+also need to use an app like [`django-storages`](https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html)
```

Comment:
au contraire, i used this at evz's suggestion for bga payroll and was so shocked at how little it took to set up that i wrote out the steps and demanded that he tell me what i was missing, lol. https://gist.github.com/hancush/7d8340d79b919fce7f0b8e6103da6e83 (i wasn't missing anything, it was that easy.)

Reply:
this was the primary issue i ran into: https://github.com/datamade/bga-payroll/issues/287

Reply:
and here is the resolution: https://github.com/datamade/bga-payroll/pull/288

Reply:
Amazing! I adjusted to note that you've piloted this already in https://github.com/datamade/how-to/pull/38/commits/4e631dec574f76c490d7f563e811a79c88101712. Your notes will be really helpful during the documentation phase of the transition.